dev015-making-apps-more-powerful/04-custom-data-final/add-custom-data/OpenExtensionsDemo.cs

```diff
@@ -1,34 +1,29 @@
-﻿using Microsoft.Identity.Client;
+﻿using device_code_flow_console;
+using Microsoft.Identity.Client;
 using Newtonsoft.Json.Linq;
 using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace add_custom_data
 {
-    class OpenExtensionsDemo
+    class OpenExtensionsDemo : PublicAppUsingDeviceCodeFlow
     {
-        PublicClientApplication app;
+        public OpenExtensionsDemo(PublicClientApplication app) : base(app)
+        {
+        }
+
         /// <summary>
         /// Scopes to request access to the protected Web API (here Microsoft Graph)
         /// </summary>
         public static string[] Scopes { get; set; } = new string[] { "User.Read", "User.ReadBasic.All", "User.ReadWrite" };
 
         public async Task RunAsync()
         {
-
-
-            AuthenticationConfig config = AuthenticationConfig.ReadFromJsonFile("appsettings.json");
-            app = new PublicClientApplication(config.ClientId, config.Authority);
             // string[] scopes = { "User.ReadWrite" };
-
-            var myInfo = new MyInformation();
-
-            AuthenticationResult authenticationResult = await myInfo.AcquireATokenFromCacheOrDeviceCodeFlow(Scopes);
+            AuthenticationResult authenticationResult = await AcquireATokenFromCacheOrDeviceCodeFlow(Scopes);
```

Comment:
here we call the base class's method, rather than delegate to an unrelated object which did not have the app initialized